clang/test/Analysis/engine/expr-engine-cxx-crash.cpp

```diff
@@ -0,0 +1,11 @@
+// RUN: %clang_analyze_cc1 -analyzer-checker=core -std=c++23 -verify %s
+// expected-no-diagnostics
+
+struct S {
+  bool operator==(this auto, S) {
+    return true;
+  }
+};
+int use_deducing_this() {
+  return S{} == S{};
```

Comment:
```suggestion
  return S{} == S{}; // no-crash
```

Reply:
updated in [f9779b3](https://github.com/llvm/llvm-project/pull/83585/commits/f9779b3a0a818b159cd9dc54fb5fd01d5b6ccaf6)